docs/environment_variables.md

```diff
@@ -32,13 +33,16 @@ AUTH_TYPE=""
 REMOTE_ADDR="127.0.0.1"
 # The server's IP address
 REMOTE_HOST="127.0.0.1"
-# The path portion of the URL. E.g. http://example.com/envwasm becomes /envwasm
-PATH_INFO="/envwasm"
+# The path info after the SCRIPT_NAME. If the route is /envwasm/... and the 
+# request is /envwasm/foo, the PathInfo is /foo. This value is run through a URL decoder
+PATH_INFO="/foo"
+# Wagi-specific: Same as PATH_INFO, but without being url-decoded. This can disambiguate
+# cases where things like '.', '+', and '/' appear in the path segment.
```

Comment:
should this be 'cases where `?`, `=`, and `;` appear in the path segment.' ?  Or are you saying that the entire string should be urlencoded , in which case should the leading `/` be encoded?

Reply:
The test cases [here](https://github.com/deislabs/wagi-dotnet/blob/2437f5ce2a5ddcfd08aef396777d3687501ea17f/source/wagi.test/WagiHostTest.cs#L85-L91) show how `X_PATH_INFO` and `X_FULL_URL` look when encoding `PATH_INFO`

Reply:
Okay, so the `?`, `=`, and `;` thing is in regard to how the PATH_INFO is _decoded_ when the incoming HTTP request is received.

Here's an example URL:

```
http://example.com/foo.cgi/bar%2bfile.txt?some=data&more=data
```

Say that `SCRIPT_NAME` is `/foo.cgi`. Now we have to parse out this according to the CGI spec: `/bar%2bfile.txt?some=data&more=data`. My reading of the spec is that it should NOT parse the `?some=data&more=data` as part of the PATH_INFO because as soon as it hits one of `?`, `=`, or `&`, it should treat the rest as query params. (Technically, the spec is incorrect on this. Only the `?` marks it as query params. = and & only take on special meaning after the ?)

Anyway... so the portion of the above that should be treated as path info is: `bar%2bfile.txt`. Now, %2b is `+`. Since `+` is allowed in a path segment, you can have a path like this: `bar%2b+file.txt` (same for `.` and `/`). The CGI spec seems to add a strange condition for `/` -- namely, that it is not supposed to appear in a not-yet-decoded PATH_INFO. The rationale is because it would be ambiguous between `/` (literal) and `%2f` (encoded `/`).

But the spec does not address the other characters, like `+` (which can be decoded to a space) or `.`, which is used for file extensions. That is why I say that I don't understand why the CGI spec prohibits path separates in raw path info, but does not prohibit other ambiguous characters. If `bar%2b+file.txt` is URL-decoded before being set to the value of `PATH_INFO`, then it will be `bar++file.txt` with no way of knowing which special chars were encoded and which were literal.

So now you need something like `RAW_PATH_INFO` to figure that out. That was the point I as trying to make in brief.

Reply:
Also, your test cases look great